File: Session14/C++/TopologicalSort.cpp
https://practice.geeksforgeeks.org/problems/topological-sort/1/

    void sorting(int node, vector<int> adj[], stack<int>& s, vector<int>& visited)
    {
        visited[node] = 1;
        for(int nbr:adj[node])
        {
            if(!visited[nbr])
            {
                sorting(nbr, adj, s, visited);
            }
        }
        s.push(node);
    }
    
	public:
	//Function to return list containing vertices in Topological order. 
	vector<int> topoSort(int V, vector<int> adj[]) 
	{
	    stack<int> s;
	    vector<int> visited(V, 0);
	    
	    for(int i=0; i<V; i++)
	    {
	        if(!visited[i])
	        {
	            sorting(i, adj, s, visited);
	        }
	    }
	    
	    vector<int> res;
	    while(!s.empty())
	    {
	        res.push_back(s.top());
	        s.pop();
	    }
	    
	    return res;
	    
	}

File: Session14/Java/TopologicalSort.java
https://practice.geeksforgeeks.org/problems/topological-sort/1/

class Solution
{
    
    static void findTopoSort(int node, int vis[], ArrayList<ArrayList<Integer>> adj, Stack<Integer> st) 
    {
        vis[node] = 1; 
        for(Integer it: adj.get(node)) 
        {
            if(vis[it] == 0) 
            {
                findTopoSort(it, vis, adj, st); 
            } 
        }
        st.push(node); 
    }
    
    //Function to return list containing vertices in Topological order. 
    static int[] topoSort(int V, ArrayList<ArrayList<Integer>> adj) 
    {
        // add your code here
        Stack<Integer> st = new Stack<Integer>(); 
        int vis[] = new int[V]; 
        
        for(int i = 0;i<V;i++) 
        {
            if(vis[i] == 0) 
            {
                findTopoSort(i, vis, adj, st);
            }
        }
        
        int topo[] = new int[V];
        int ind = 0; 
        while(!st.isEmpty()) 
        {
            topo[ind++] = st.pop();
        }

        return topo; 
    }
}